src/pymordemos/linear_optimization.py

```diff
@@ -0,0 +1,151 @@
+    opt_rom_minimization_data['time'] = perf_counter()-tic
+
+    print("\nResult of optimization with FOM model and adjoint gradient")
+    report(opt_fom_result, fom.parameters.parse, opt_fom_minimization_data, reference_mu)
+    print("Result of optimization with ROM model and adjoint gradient")
+    report(opt_rom_result, fom.parameters.parse, opt_rom_minimization_data, reference_mu)
+
+def create_fom(grid_intervals, vector_valued_output=False):
+    domain = RectDomain(([-1,-1], [1,1]))
+    indicator_domain = ExpressionFunction(
+        '(-2/3. <= x[..., 0]) * (x[..., 0] <= -1/3.) * (-2/3. <= x[..., 1]) * (x[..., 1] <= -1/3.) * 1. \
+       + (-2/3. <= x[..., 0]) * (x[..., 0] <= -1/3.) *  (1/3. <= x[..., 1]) * (x[..., 1] <=  2/3.) * 1.',
+        dim_domain=2, shape_range=())
+    rest_of_domain = ConstantFunction(1, 2) - indicator_domain
+
+    f = ExpressionFunction('0.5*pi*pi*cos(0.5*pi*x[..., 0])*cos(0.5*pi*x[..., 1])', dim_domain=2, shape_range=())
+
+    parameters = {'diffusion': 2}
+    thetas = [ExpressionParameterFunctional('1.1 + sin(diffusion[0])*diffusion[1]', parameters,
+                                       derivative_expressions={'diffusion': ['cos(diffusion[0])*diffusion[1]',
+                                                                             'sin(diffusion[0])']}),
+          ExpressionParameterFunctional('1.1 + sin(diffusion[1])', parameters,
+                                       derivative_expressions={'diffusion': ['0',
+                                                                             'cos(diffusion[1])']}),
+
+                                       ]
+    diffusion = LincombFunction([rest_of_domain, indicator_domain], thetas)
+
+    theta_J = ExpressionParameterFunctional('1 + 1/5 * diffusion[0] + 1/5 * diffusion[1]', parameters,
+                                           derivative_expressions={'diffusion': ['1/5','1/5']})
+
+    if vector_valued_output:
+        problem = StationaryProblem(domain, f, diffusion, outputs=[('l2', f * theta_J), ('l2', f *0.5* theta_J)])
+    else:
+        problem = StationaryProblem(domain, f, diffusion, outputs=[('l2', f * theta_J)])
+
+    print('Discretize ...')
+    mu_bar = problem.parameters.parse([np.pi/2,np.pi/2])
+    fom, _ = discretize_stationary_cg(problem, diameter=1. / grid_intervals, mu_energy_product=mu_bar)
+    return fom, mu_bar
+
+def record_results(function, parse, data, mu):
+    QoI = function(mu)
+    data['num_evals'] += 1
+    # we need to make sure to copy the data, since the added mu will be changed inplace by minimize afterwards
+    data['evaluation_points'].append([parse(mu)['diffusion'][:][0],
+                                      parse(mu)['diffusion'][:][1]])
+    data['evaluations'].append(QoI[0])
+    print('.', end='')
+    return QoI
+
+def report(result, parse, data, reference_mu=None):
+    if (result.status != 0):
+        print('\n failed!')
+    else:
+        print('\n succeded!')
```

Comment:
```suggestion
        print('\n succeeded!')
```

Reply:
Will be fixed in #1110

Reply:
Oh yeah, sorry, I did not recognize that this was not the `.rst` file anymore.